File: Web/RMA-Web-Shell/projects/clientcare/src/app/member-manager/models/industry-class-declaration-configuration.ts
import { IndustryClassEnum } from 'projects/shared-models-lib/src/lib/enums/industry-class.enum';
import { DeclarationPenaltyPercentage } from './declaration-penalty-percentage';
import { MaxAverageEarning } from './max-average-earning';
import { LiveInAllowance } from './live-in-allowance';
import { InflationPercentage } from './inflation-percentage';
import { MinimumAllowablePremium } from './minimum-allowable-premium';

export class IndustryClassDeclarationConfiguration {
    industryClassDeclarationConfigurationId: number;
    industryClass: IndustryClassEnum;
    renewalPeriodStartMonth: number;
    renewalPeriodStartDayOfMonth: number;
    varianceThreshold: number;
    onlineSubmissionStartMonth: number;
    onlineSubmissionStartDayOfMonth: number;

    maxAverageEarnings: MaxAverageEarning[];
    declarationPenaltyPercentages: DeclarationPenaltyPercentage[];
    liveInAllowances: LiveInAllowance[];
    inflationPercentages: InflationPercentage[];
    minimumAllowablePremiums: MinimumAllowablePremium[];
}

File: Service/ClientCare/RMA.Service.ClientCare.Contracts/Entities/Member/IndustryClassDeclarationConfiguration.cs
using RMA.Service.Admin.MasterDataManager.Contracts.Enums;

using System.Collections.Generic;

namespace RMA.Service.ClientCare.Contracts.Entities.Member
{
    public class IndustryClassDeclarationConfiguration
    {
        public int IndustryClassDeclarationConfigurationId { get; set; }
        public IndustryClassEnum IndustryClass { get; set; }
        public int RenewalPeriodStartMonth { get; set; }
        public int RenewalPeriodStartDayOfMonth { get; set; }
        public decimal? VarianceThreshold { get; set; }
        public int? OnlineSubmissionStartMonth { get; set; }
        public int? OnlineSubmissionStartDayOfMonth { get; set; }

        public List<MaxAverageEarning> MaxAverageEarnings { get; set; }
        public List<DeclarationPenaltyPercentage> DeclarationPenaltyPercentages { get; set; }
        public List<LiveInAllowance> LiveInAllowances { get; set; }
        public List<InflationPercentage> InflationPercentages { get; set; }
        public List<MinimumAllowablePremium> MinimumAllowablePremiums { get; set; }
    }
}